### DISCUSSION TITLE
Register controllers, models, validators in provider

### DISCUSSION THREAD

**Discussion by sjorobekov** [adonisjs/core#4176] [Help] [answered]
Hi there,

I want to make "installable" module, so wondering if it is possible to register all these things in provider or is there any other option?

Thanks in advance

**Comment by MaximeMRF** [comment, marked as answer]
Hello @sjorobekov 

You want to create a Adonisjs package ?

This [video](https://youtu.be/9N-OoVW9dDk) explain how to develop a package for Adonis. I think it also cover the publishing of the package.

You can extends some class like [Request](https://docs.adonisjs.com/guides/request#extending-request-class), [Router](https://docs.adonisjs.com/guides/routing#extending-router) ect.. or wrap a node lib, depend of what do you want to do.
